tests/test_dfes.py

```diff
@@ -0,0 +1,167 @@
+            mutation_types=[dfe.MutationType(convert_to_substitution=True)],
+            proportions=[1.0],
+        )
+        ds = str(d)
+        expected = """\
+        DFE:
+        ║  id               = xyz
+        ║  description      = abc
+        ║  long_description = ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC
+        ║                     ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC
+        ║                     ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC ABC
+        ║  citations        = []
+        """
+        assert textwrap.dedent(expected) in ds
+
+
+class TestRegisterQCDFE:
+    """
+    Tests for registering a QC DFE.
+    """
+
+    def make_dfe(self, name):
+        return dfe.DFE(
+            id=name,
+            description=name,
+            long_description=name,
+            mutation_types=[dfe.MutationType(convert_to_substitution=True)],
+            proportions=[1.0],
+        )
+
+    def test_register_qc(self):
+        dfe = self.make_dfe("test")
+        dfe.register_qc(dfe)
+
+    def test_already_registered(self):
+        dfe = self.make_dfe("test")
+        dfe.register_qc(dfe)
+        with pytest.raises(ValueError):
+            dfe.register_qc(dfe)
+
+    def test_bad_qc_dfe(self):
+        dfe = self.make_dfe("test")
+        for not_a_dfe in [None, 15, "Zigzag_1S14"]:
+            with pytest.raises(ValueError):
+                dfe.register_qc(not_a_dfe)
+
+
+class DFETestMixin:
+    """
+    Mixin for testing specific DFEs. Subclass should extend
+    this class and define the self.dfe (as the dfe instance).
+    """
+
+    dfe = None
+
+
+class CatalogDFETestMixin(DFETestMixin):
+    """
+    Mixin for DFEs in the catalog.
+    """
+
+    def test_id_valid(self):
+        assert utils.is_valid_dfe_id(self.dfe.id)
+
+    #def test_params_match_docs_tables(self):
+    #    for species in stdpopsim.all_species():
+    #        for d in species.dfes:
+    #            table_path = pathlib.Path()
+    #            if d.qc_dfe is not None:
+    #                assert table_path.exists()
+    #                with 
+
+
+class QcdCatalogDFETestMixin(CatalogDFETestMixin):
+    """
+    Extends the tests to also check that the qc DFE is equal to
+    the production DFE.
+    """
+
+    def test_mutation_types_match(self):
+        mt1 = self.dfe.mutation_types
+        mt2 = self.dfe.qc_dfe.mutation_types
+        assert len(mt1) == len(mt2)
+
+        for i in range(len(mt1)):
+            assert mt1[i].dominance_coeff == mt2[i].dominance_coeff
+            assert mt1[i].distribution_type == mt2[i].distribution_type
+            assert mt1[i].distribution_args == mt2[i].distribution_args
+            assert mt1[i].convert_to_substitution ==  mt2[i].convert_to_substitution
+
+    def test_proporitions_match(self):
+        p1 = self.dfe.proportions
+        p2 = self.dfe.qc_dfe.proportions
+        assert p1 == p2
```

Comment:
Shouldn't this be `assert np.all(p1 == p2)`? Or, `assert np.allclose(p1 == p2)`?

Reply:
changed to `assert np.allclose(p1, p2)`